client/src/components/main/chatwindow/ChatWindow.js

```diff
@@ -0,0 +1,33 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import CreateComment from './CreateComment';
+import ViewComments from './ViewComments';
+
+/*
+Parent component
+State is a boolean that changes when a new comment has been created and rendered
+*/
+
+const ChatWindow = props => {
+  const [newComment, setNewComment] = useState(false);
```

Comment:
What is `[newComment, setNewComment]` and what is it supposed to do?

Reply:
I use `[newComment, setNewComment]` to track whether a new comment has been submitted by passing them down as props, triggering a re-render. It's in the parent component because both children components need to access the value. `CreateComment` uses `setNewComment` to change state, which initiates a re-render in `ViewComments`. Within `useEffect`, `ViewComments` uses `setNewComment` to change state again to trigger a second-rerender since the get request has a delay. When I was only changing state once through `setAllComments` in `ViewComments`, the response did not contain the new comment. I used the boolean `newComment` to trigger a second re-render, essentially a second get request to account for the delay.

Reply:
I've removed this in favor of `useReducer`